### PR TITLE
Add ability to load checkpoints from cli

### PR DESCRIPTION
**Pull request by randombenj**
## What does this PR do?

**What does it do?**

This PR adds the ability to load checkpoints from the Lightning CLI. At the moment, checkpoint loading
is limited to the model (trainer, data loader could be added easily if requested).

**Why is this change important?**

I think the lightningcli is a fantastic tool to manage different trainer/model/dataloader configurations.
Especially that the model params are loaded from `yaml` config files.

In my case I would like to export the trained models to `onnx` or `torchscript` with the same hyperparameter config 
(loaded by lightningcli).

```python
# run with --load_from_checkpoint path/to/checkpoint.ckpt
cli = LightningCLI(run=False)
cli.model.to_onnx("model.onnx")
```

In the future it would also be nice to load trainer and dataloaders from checkpoints to continue training ...

### Does your PR introduce any breaking changes? If yes, please list them.

No

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by randombenj** [comment]
After looking through some PR's I think the failing tests are not due to my change (I might be mistaken however).
Even the main branch seems to not have passing checks for quite some time now :)

**Comment by carmocca** [comment]
Any thoughts on this @mauvilsa?

> In the future it would also be nice to load trainer and dataloaders from checkpoints to continue training ...

This cannot be easily done as they are supposed to be reloaded when the `Trainer` starts running

**Comment by randombenj** [comment]
@carmocca Well maybe not for the dataloaders but `ckpt_path` could also be added to the `trainer.fit` function (according to the docs) https://pytorch-lightning.readthedocs.io/en/stable/common/checkpointing.html#checkpoint-loading.

This could enable interrupting and continuing training in via LightningCLI.

Another idea would be to  add an `export` subcommand to https://github.com/PyTorchLightning/pytorch-lightning/blob/fd14bbb7e44b77eb8500326e22e1de18e2ea1d6d/pytorch_lightning/utilities/cli.py#L661 to easily export a trained model with weights to `onnx/torchscript/...` (forwarding sample data for tracing could then be another feature, maybe taken from the dataloader, when `Module.example_input_array` is not defined in the Module to export

Or also just run inference with pretrained weights using LightningCLI ...

**Comment by mauvilsa** [comment]
Adding globally an `--load_from_checkpoint` option means that it will be available for all subcommands and instantiation only, i.e. `run=False`. But the subcommands already have the `ckpt_path` parameter, which would make it redundant and confusing.

The idea of having a subcommand for exporting makes more sense to me. This would be for a use case in which someone trains a bunch of models and only wants to export one or a few of them. If the use case is that always when a model is fit it should also be exported, then this is more for the user to add to `after_fit` or in a callback that implements `on_fit_end`.

> Or also just run inference with pretrained weights using LightningCLI ...

There is already the `predict` subcommand for this, no? This might have some relation to the discussion in #12438.

**Comment by randombenj** [comment]
> Adding globally an --load_from_checkpoint option means that it will be available for all subcommands and instantiation only, i.e. run=False. But the subcommands already have the ckpt_path parameter, which would make it redundant and confusing.

I was not aware that one could use `ckpt_path` for the subcommands, but I do agree that two checkpoint options do not make sense here. RTFM I guess :) 

> The idea of having a subcommand for exporting makes more sense to me. This would be for a use case in which someone trains a bunch of models and only wants to export one or a few of them. If the use case is that always when a model is fit it should also be exported, then this is more for the user to add to `after_fit` or in a callback that implements `on_fit_end`.

Should close this PR then and maybe add an `export` subcommand as this would cover my use case. But wouldn't you anyway something similar like this PR to only load the model weights for exporting? I guess one shouldn't instantiate a trainer just to export the model?
